### ISSUE TITLE
can't pickle PDBReader on python 3

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#1981]
### Expected behaviour
I'm able to pickle PDBReader under python 3

### Actual behaviour

I'm getting an error that TextIOWrapper cannot be pickled.

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda
from MDAnalysisTests.datafiles import PDB
import pickle

u = mda.Universe(PDB, PDB)
pickle.dumps(u.trajectory)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-d286063031e3> in <module>()
----> 1 pickle.dumps(u.trajectory)

TypeError: cannot serialize '_io.TextIOWrapper' object

....

```

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
0.18.0

**Comment by orbeckst** [comment]
See #1940 – this issue is part of the problem.

**Comment by orbeckst** [comment]
I am labeling this as a defect because this is supposed to work (and will actually break code that tries to parallelize anything that reads a PDB!)

**Comment by kain88-de** [comment]
the chainreader problem seems to be a different one. I actually wanted to see where it uses generators and then came across this problem. 

I should note that it works to pickle with cloudpickle

**Comment by orbeckst** [comment]
> I should note that it works to pickle with cloudpickle

Can you please be more precise? What works with cloudpickle?

**Comment by orbeckst** [comment]
The generator is https://github.com/MDAnalysis/pmda/issues/44#issuecomment-397485426
```
ChainReader__chained_trajectories_iter': <generator object ChainReader._chained_iterator at 0x7f1c19716468>
```

> the chainreader problem seems to be a different one

I said it is part of the problem because I am pretty sure that as soon as we solve the generator problem, we run into the issue described by you.

**Comment by kain88-de** [comment]
You can pickle the PDB reader with the cloudpickle library. It’s a drop in replacement for pickle.

**Comment by orbeckst** [comment]
closed with PR #2723
